### ISSUE TITLE
I wish /api-ref/ existed, because the K8s API is practically unusable. It is so entwined with the labyrinth of K8s packages.

### ISSUE THREAD

**Issue by r1ky** [kubernetes/kubernetes.github.io#165]
I wish /api-ref/ existed, because the K8s API is practically unusable by outside parties. It is so entwined with the labyrinth of K8s packages.

**Comment by johndmulhausen** [comment]
I'm confused. Do you mean http://kubernetes.io/docs/api/ ?
See the "Operations" and "Definitions" page for reference docs on the API. The reference section also has docs for Extensions API, and all the CLIs.

**Comment by r1ky** [comment]
Hey John - can’t even remember the link I hit. But I got taken to the 404 page which prompted me to create the issue. 

I was being a little snarky I admit when I used to opportunity to gripe about the API. But I have to say the Kubernetes API really is pretty much unusable by 3rd parties in it’s current state. I see there is an open ticket about reducing it’s footprint. And I would certainly support that.

My use case is to avoid kubectl | grep | awk in my deployment scripts to get various bits and pieces of data concerning our svc, rc and pods. ATM just internal IP address and ports are needed. So the idea of compiling all of Kubernetes (almost) so I can write a dinky API request to get some JSON to unmarshall is a non-starter.

- Rick
